src/preact/index.js

```diff
@@ -139,6 +139,17 @@ export function useCallback(cb, opt_deps) {
   return hooks.useCallback(cb, opt_deps);
 }
 
+/**
+ * @param {{current: (T|null)}} ref
```

Comment:
Nit: this could also be a function. But not sure if either is ever really observed by the caller - it's just passed through.

Reply:
You mean `{{current: (T|function|null)}}`?

Reply:
No. It's `{{current: T|null}|function(T|null)}`. I.e. an object or a function.

Reply:
Do you have an example of a ref behaving as a function? Is this something set via `useImperativeHandle` or given to it?

Reply:
A function can be used anywhere where a ref is used. E.g.

```
const ref = useRef();
<div ref={ref}>
```
vs

```
<div ref={(value) => console.log('new ref value is resolved: ', value)}>
```

And the same with `useImperativeHandle`.

But, yes, you're right - as an author you'd not be interested in what it is - it's just an opaque thing you pass to the `useImperativeHandle`.

Reply:
But IMHO, feel free to start with a simpler form and we can wait until we run into a function-based use case.